api/tests/src/task/opportunities/test_store_opportunity_version_task.py

```diff
@@ -0,0 +1,101 @@
+import pytest
+from sqlalchemy import delete
+
+from src.constants.lookup_constants import JobStatus
+from src.db.models.opportunity_models import Opportunity, OpportunityChangeAudit, OpportunityVersion
+from src.db.models.task_models import JobLog
+from src.task.opportunities.store_opportunity_version_task import StoreOpportunityVersionTask
+from tests.conftest import BaseTestClass
+from tests.src.db.models.factories import (
+    JobLogFactory,
+    OpportunityChangeAuditFactory,
+    OpportunityVersionFactory,
+)
+
+
+class TestStoreOpportunityVersionTask(BaseTestClass):
+    @pytest.fixture
+    def store_opportunity_version_task(self, db_session):
+        return StoreOpportunityVersionTask(db_session)
+
+    @pytest.fixture(autouse=True)
+    def clear_db(self, db_session):
+        opportunities = db_session.query(Opportunity).all()
+        for opp in opportunities:
+            db_session.delete(opp)
+
+        db_session.execute(delete(OpportunityVersion))
+        db_session.execute(delete(OpportunityChangeAudit))
```

Comment:
Minor nitpick, technically those two deletes do nothing because they had to have been deleted for the opportunity deletes to work (the `cascade="all, delete-orphan"` on the relationships tells SQLAlchemy to cleanup orphaned records if the opportunity itself is deleted).

```suggestion
    @pytest.fixture(autouse=True)
    def clear_db(self, db_session):
        opportunities = db_session.query(Opportunity).all()
        for opp in opportunities:
            db_session.delete(opp)
```

Reply:
Thats right ! Deleted